README.md

```diff
@@ -1 +1,209 @@
+#### `common.proto` - Base Protocol Definitions
+- `Error`: Error reporting with cause description
+- `Close`: Channel closure signaling
+- `DataChunk`: Raw data chunk for streaming
+- `SendingMessage`: Message sent from runner to orchestrator with channel, data, and local sequence number
+- `ReceivingMessage`: Message received by runner from orchestrator with global sequence number, channel, and data
+- `ReceivingStreamMessage`: Stream message received by runner with global sequence number and channel
+- `ReceivingStreamControl`: Control message for streaming to producer (contains stream sequence number)
+- `SendingStreamControl`: Control message for streaming from consumer (contains global or stream sequence number)
+- `StreamIdentify`: First message when runner sends streaming data (channel, local sequence number, runner URI)
+- `StreamChunk`: Wrapper for StreamIdentify or DataChunk messages
+- `MessageProcessed`: Processing acknowledgment with channel and global sequence number
+
+#### `service.proto` - Service Interface
+- `LogMessage`: Structured logging with levels, entities, and aliases
+- `Processor`: Processor definition with URI, configuration, and arguments
+- `ToRunner`: Message container sent to runners (contains processor, start, message, close, stream message, pipeline, or processed)
+- `ProcessorInitialized`: Initialization confirmation with URI and optional error
+- `RunnerIdentify`: Runner identification with URI
+- `FromRunner`: Message container from runners (contains initialized, close, identify, message, or processed)
+- `Runner` Service: Main service interface with 4 RPC methods
+
+
+## Message Flow Diagrams
+
+### Startup Sequence
+
+```mermaid
+sequenceDiagram
+    participant R as Runner
+    participant O as Orchestrator
+
+    Note over R,O: Connection Establishment
+    R->>O: connect() - Initiate bidirectional stream
+
+    Note over R,O: Identify
+    R->>O: FromRunner{identify: RunnerIdentify(uri)}
+
+    Note over O,R: Pipeline Updates
+    O->>R: ToRunner{pipeline: "Turtle RDF with SHACL shapes"}
+    R->>R: Update processing pipeline
+
+    Note over R,O: Processor Configuration
+    O->>R: ToRunner{proc: Processor(uri, config, arguments)}
+
+    Note over R,O: Initialization Confirmation
+    R->>O: FromRunner{initialized: ProcessorInitialized(uri, [error])}
+
+    Note over R,O: Processing Start
+    O->>R: ToRunner{start: Empty}
+    R->>R: Ready for processing
+```
+
+### Message Processing Flow
+
+```mermaid
+sequenceDiagram
+    participant R2 as Runner2
+    participant O as Orchestrator
+    participant R1 as Runner1
+
+    Note right of O: Standard Message Flow
+    R1->>O: FromRunner{SendingMessage(localSequenceNumber, channel, data)}
+    O->>R2: ToRunner{msg: ReceivingMessage(globalSequenceNumber, channel, data)}
+    R2->>O: FromRunnner{processed: MessageProcessed(channel, globalSequenceNumber)}
+    O->>R1: ToRunner{processed: MessageProcessed(channel, globalSequenceNumber: localSequenceNumber)}
```

Comment:
Is this meant to be like this? global...: local...
Idem in streaming message flow diagram.

Reply:
Right, MessageProcessed just says globalSequenceNumber, but it is the localsequence number that is filled in :(

Reply:
Yes, but first R2 will send a MessageProcessed with the globalSequenceNumber to the orchestrator to indicate that message is handled

Reply:
So we _could_ split the MessageProcessed up in two as we did with the sending and receiving message, but I'm clueless to the naming

Reply:
I don't really care, it's mostly naming that holds me back